packages/arb-token-bridge-ui/src/hooks/useNetworksAndSigners.tsx

```diff
@@ -156,18 +156,38 @@ export function NetworksAndSignersProvider(
     async (web3Provider: Web3Provider, address: string) => {
       const providerChainId = (await web3Provider.getNetwork()).chainId
 
+      // If provider is not supported, display warning message
+      if (!(providerChainId in chainIdToDefaultL2ChainId)) {
+        console.error(`Provider chainId not supported: ${providerChainId}`)
+        setResult({ status: UseNetworksAndSignersStatus.NOT_SUPPORTED })
+      }
+
+      /***
+       * Case 1: selectedChainId is undefined => set it to provider's default L2
+       * Case 2: selectedChainId is defined but not supported by provider => reset query params -> case 1
+       * Case 3: selectedChainId is defined and supported, continue
+       */
       let _selectedL2ChainId = selectedL2ChainId
+      const providerSupportedL2 = chainIdToDefaultL2ChainId[providerChainId]
+
+      // Case 1: use a default L2 based on the connected provider chainid
+      _selectedL2ChainId = _selectedL2ChainId || providerSupportedL2[0]
```

Comment:
I think we can do this without a need for reassignment and with some destructuring:

```suggestion
const [defaultL2ChainId] = chainIdToDefaultL2ChainId[providerChainId]
// Case 1: use a default L2 based on the connected provider chain id
const _selectedL2ChainId = selectedL2ChainId ?? defaultL2ChainId
```

Reply:
We still need to access `chainIdToDefaultL2ChainId[providerChainId]` few lines [below](https://github.com/OffchainLabs/arb-token-bridge/pull/429/files#diff-9d4f6711941cc58931f96b21b6d6ffa98e2e501dbddbd5fe57592037422d76bdR182) which is why we have `providerSupportedL2` as a variable.